### ISSUE TITLE
Wrong async bundle reference

### ISSUE THREAD

**Issue by btamas** [lasso-js/lasso#85]
I have a project that is written in es6 with commonjs and I use lasso to build the bundles.
I have some async bundles, that are defined in lasso-build.json config file like:

```
...
"bundles": [
    {
      "name": "proto1",
      "dependencies": [
        "require: ../src/layouts/proto1/index.es6"
      ]
    },
    {
      "name": "proto2",
      "dependencies": [
        "require: ../src/layouts/proto2/index.es6"
      ]
    },
...
```

I load the modules like this way:

```
loadProto1(callback) {
		require('raptor-loader').async(() => {
			import proto1 from '../layouts/proto1/index.es6';
			this.layout = proto1;
			callback(this.layout);
		});
	}
```

I got a following error message in browser:
Uncaught Error: Cannot find module "../layouts/proto1/index.es6" from "/src/lib"

The generated code that wants to load that bundle is:
```
loadProto1 = function loadProto1(callback) {
		var _this = this;

		require('raptor-loader'/*'raptor-loader'*/).async("_6", function () {
			var _layoutsProto1IndexEs6 = require('../layouts/proto1/index.es6');

			var _layoutsProto1IndexEs62 = _interopRequireDefault(_layoutsProto1IndexEs6);

			_this.layout = _layoutsProto1IndexEs62['default'];
			callback(_this.layout);
		});
	};
```

It's looking good, but the problem is that the module is not in the bundle named '_6', because it is in the '_8'.

My $rloaderMeta is the following:
```
{
...
  "_6": {
    "js": ["http://{domain}/js/live.js","http://{domain}/js/default-async.js"]
  },
...
  "_8": {
    "js":["http://{domain}/js/proto1.js"]
  }
...
}
```

In the beginning I had the same problem, but when I changed the order of the definition of the bundles in the lasso-build config, it worked. But now I cannot find any order that can work.

**Comment by philidem** [comment]
Hey @btamas. Can you try deleting your `.cache` directory at the root of the project and re-running the lasso bundler? I suspect that this is a caching issue and I have seen it before but it is hard to reproduce. Obviously, this is not a great workaround so I think we should revisit some of the caching logic if this does fix your problem.

**Comment by btamas** [comment]
@philidem This was the first thing that I tried, but it's not working.
Now I'm trying to find where is the problem, but the code is very complex and everything is looking good, so I do not know what should I check.

**Comment by patrick-steele-idem** [comment]
Hey @btamas, that does look like a bug and I am pretty sure it is due to the `.es6` extension being used in the require path. Lasso normalizes the paths by removing the file extension, but it looks like that might not be happening correctly with the `.es6` extension. I'm going to investigate. Thanks for reporting the issue.

**Comment by patrick-steele-idem** [comment]
Actually, after reading over it again I may be wrong. Either way, I'll investigate :)

**Comment by patrick-steele-idem** [comment]
Hey @btamas, I'm still investigating but when I use the following code:

```javascript
    loadProto1(callback) {
        require('raptor-loader').async(() => {
            import proto1 from '../layouts/proto1/index.es6';
            this.layout = proto1;
            callback(this.layout);
        });
    }
``` 

I get the following error:

```
SyntaxError: /Users/psteeleidem/development/github/lasso-js-samples/lasso-es6-async/src/client.es6: 'import' and 'export' may only appear at the top level (38:8)
  36 | function loadProto1(callback) {
  37 |     require('raptor-loader').async(() => {
> 38 |         import proto1 from '../layouts/proto1/index.es6';
     |         ^
  39 |         console.log('proto1 returned value: ', proto1());
  40 |     });
  41 | }
```

I'm curious as to why it is working for you. Any thoughts?

**Comment by btamas** [comment]
Did you enabled the babel plugin?
In the lasso-bulid config file you should insert something like this:
```
...
"plugins": [
    {
      "plugin": "lasso-require",
      "config": {
        "babel": {
          "blacklist": "strict",
          "loose": "all"
        }
      }
    },
...
```

**Comment by patrick-steele-idem** [comment]
Hey @btamas, I updated the code to be the following: 

```javascript
    loadProto1(callback) {
        require('raptor-loader').async(() => {
            var proto1 require('../layouts/proto1/index.es6');
            this.layout = proto1;
            callback(this.layout);
        });
    }
```

After that change, everything is working correctly and I was not able to reproduce the problem.

**Comment by patrick-steele-idem** [comment]
> Did you enabled the babel plugin?
> In the lasso-bulid config file you should insert something like this:

No, let me give that a try.

**Comment by patrick-steele-idem** [comment]
I enabled the babel config you provided and switched back to using the nested import and things are still working as expected. I'll push the code so that you can take a look. Give me just a minute

**Comment by patrick-steele-idem** [comment]
Hey @btamas, can you please take a look at the following test app that was created for this issue?: https://github.com/patrick-steele-idem/lasso-issue-85

I'm not able to reproduce the issue. Can you please see if you can tweak that app to reproduce the problem that you are seeing?

**Comment by btamas** [comment]
It's working for me also. Tomorrow I will do it, because I'm from Europe and it is evening now. Thanks for the help, and I will try to reproduce my problem in your environment.

**Comment by btamas** [comment]
I tried to reproduce it in your environment, but I cannot do it. After 3 hours I found a strange thing it my environment. When I modified the files it does not do that I wrote in them. Then I find the problem and now it's working correctly. I screwed some config and the asynchronous files were downloaded from another server that I generated before, and of course the file IDs were not correct.
Thank you for your help and really sorry for spent your time for nothing.
